packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -436,6 +436,20 @@ class AndroidDevice extends Device {
     runCheckedSync(cmd);
     return true;
   }
+
+  @override
+  bool get supportsScreenshot => true;
+
+  @override
+  Future<bool> takeScreenshot(File outputFile) {
+    const String remotePath = '/data/local/tmp/screen.png';
```

Comment:
We already rely on `/data/local/tmp` existing and being read/write by the shell user in order to launch apps.